### ISSUE TITLE
Ignoring users should remove them from the room members list

### ISSUE THREAD

**Issue by stoically** [vector-im/riot-web#12576]
When ignoring someone they'll still show up in the room members list. I'd expect them to not get shown.

And given the "right" display name, pretty much on top of the list, which can be problematic if the reason to ignore is an offensive profile picture.

**Comment by jugglinmike** [comment]
@stoically Is there currently any other location that displays ignored users? If not, it seems like we'll want to add one--otherwise, it won't be possible to "unignore" a user.

**Comment by t3chguy** [comment]
Settings > Security & Privacy > Ignored users

![image](https://user-images.githubusercontent.com/2403652/86489055-fa2a3d80-bd5a-11ea-9170-94d7234cafae.png)

**Comment by jugglinmike** [comment]
Clearly, I wasn't looking hard enough. Thanks for the help, @t3chguy. I'll give this a shot.

**Comment by jugglinmike** [comment]
If we filter out the names of ignored members without any other change to the current UI, users won't be aware of all recipients of their messages. I'm still getting a handle on Matrix's privacy model (where I know room history is available to new members, at least in some contexts), but it seems like this could subvert people's expectations.

Does that sound right? Could we address this with another "overflow" tile--something like, "and 23 ignored..."?

**Comment by t3chguy** [comment]
So ignore is actually handled server-side and already won't send you events from that user

**Comment by jugglinmike** [comment]
My concern is about whether the people I have ignored will receive my messages (not the other way around). Learning that "ignoring" is part of the protocol gives me confidence that this has already been considered in depth, but [I don't see this mentioned in the spec](https://matrix.org/docs/spec/client_server/latest#id144), so I assume the people I ignore can read my messages.

Again, I may be thinking about this incorrectly. Do Matrix users have any expectation about who can (or will) read messages they send to a room? If so, then it seems like we'd want to tell users, "there are more people in this room than you can see in the member list."

**Comment by aaronraimist** [comment]
Yes they can read your messages. The only thing ignore does is stop you from seeing their messages. There is no way to stop sending messages to a particular person. Messages are sent into rooms and anyone who is allowed to view messages in that room can see them.

**Comment by jugglinmike** [comment]
Got it. Thanks for bearing with me on this!
